### DISCUSSION TITLE
Plugin for scanning/listing virtual environments that use pyenv-managed python installations

### DISCUSSION THREAD

**Discussion by pfheatwole** [pyenv/pyenv#1783] [Show and tell]
I was tidying my work environment and wanted to know which pyenv versions were safe to uninstall. Virtual environments can be created in a variety of ways (`poetry install`, `python -m venv <name>`, etc), so it's not trivial to track which versions of pyenv-managed `python` installations are linked to virtual environments, so I threw this together:

https://gist.github.com/pfheatwole/a298e5698eeb517f770b68c154b9682c

Example output:

```
$ pyenv users ~
3.7.9          /home/peter/.cache/pypoetry/virtualenvs/my_project-KM_3YcvM-py3.7
3.7.9          /home/peter/work/venvs/long name with spaces
3.8.6          /home/peter/.cache/pypoetry/virtualenvs/my_project-KM_3YcvM-py3.8
pypy3.6-7.3.1  /home/peter/work/venvs/example1
```

Maybe there's an easier way to do this, but I thought this might help someone. Cheers!

**Comment by anton-petrov** [comment]
As for me, this is quite an interesting functionality. Maybe you should create a PR?

**Reply by pfheatwole** [reply]
I'd be happy to share, but under what repo? I can't open a PR to create a new repo like `pyenv/pyenv-users`. For that matter, I'm not even sure `users` is a great choice for the command name. :rofl:

Also, it's probably still pretty raw. The tabular output is currently using the `column` program from the `util-linux` package. Users on MacOS and Windows would probably prefer something nicer than the fallback format of `<version>:<path>`.

All that said, I'm open to recommendations. :smile: I'm in crunch mode for my main project so I will be slow to respond, but I'll do what I can.

**Reply by pfheatwole** [reply]
> As for me, this is quite an interesting functionality. Maybe you should create a PR?

I reviewed "Authoring Plugins" and realized what you're probably wanting, so I moved the gist to a repo: https://github.com/pfheatwole/pyenv-users

I'm not in a position to maintain it, but I'd be happy transferring ownership to @pyenv. Feel free to fork it, or I can open a PR at https://github.com/pyenv/pyenv/issues

Cheers! :beers:

**Reply by yyuu** [reply]
I think making @anton-petrov as admin of @pyenv will give him more flexibility. Also, I can add @pfheatwole as a plugin maintainer of the org. Give me a minute :pray:

**Reply by yyuu** [reply]
Now it should be all set :tada: https://github.com/pyenv/pyenv-users

**Reply by pfheatwole** [reply]
Nice! I've archived my repo with a link to the new home. I also went ahead and pushed a fix for detecting environments owned by `pyenv-virtualenv`. I can't commit to being a regular maintainer, but I'll do what I can.

Also, for this plugin: I'm still not sure "users" is a good command name. If either of you think it should change then feel free. Maybe a verb, like "list-users"? I have no preference.

**Reply by anton-petrov** [reply]
Thanks, @yyuu!  

As for the name of the command (plugin) - suggest to think it over, personally, I'm for `pyenv-users` too. It's a good obvious name.
